tests/memory-report.cpp

```
@@ -0,0 +1,349 @@
+}
+
+void writeMemoryReport()
+{
+    const std::string& path = options().memoryReportFile;
+    if (!options().memoryReport || path.empty())
+        return;
+
+    std::filesystem::path reportPath(path);
+    if (reportPath.has_parent_path())
+        std::filesystem::create_directories(reportPath.parent_path());
+
```

Cppcheck (2.21.0)

[style] The function 'swapWith' is never used.

(unusedFunction)